cmd/lotus/daemon.go

```diff
@@ -269,7 +269,37 @@ var DaemonCmd = &cli.Command{
 			}
 		}
 
-		if cctx.Bool("remove-existing-chain") {
+		chainfile := cctx.String("import-chain")
+		snapshot := cctx.String("import-snapshot")
+		willImportChain := false
+		if chainfile != "" || snapshot != "" {
+			if chainfile != "" && snapshot != "" {
+				return fmt.Errorf("cannot specify both 'import-snapshot' and 'import-chain'")
+			}
+			willImportChain = true
+		}
+
+		willRemoveChain := cctx.Bool("remove-existing-chain")
+		if willImportChain && !willRemoveChain {
+			// Confirm with the user about the intention to remove chain data.
+			reader := bufio.NewReader(os.Stdin)
+			fmt.Print("Importing chain or snapshot will by default delete existing local chain data. Do you want to proceed and delete? (yes/no): ")
```

Comment:
```suggestion
			fmt.Print("Do you want to delete existing local chain data as part of the import process? (yes/no): ")
```

Reply:
(just a bit more clear, there isn't really a default since we're forcing users to specify)

Reply:
You are right, however we couldn't differentiate between
- user not specifying a flag value
- user specifically specifying a flag to false

That's why we have to ask for the user's input...